### ISSUE TITLE
running_instances = -1 for failed staging

### ISSUE THREAD

**Issue by KrisKobylinski** [cloudfoundry/cloud_controller_ng#406]
After pushing an application which fails staging because of missing buildpack support and pulling the application state afterwards the following parameters can be observed:
"state": "STARTED"
"running_instances":-1
"package_state": “FAILED"

matthew.sykes@gmail.com said:

While the '-1' does imply that hm9k doesn't know about the app, a stale view is not the only cause.

The problem that you're hitting is that the summary endpoint isn't doing the right thing when the app has not been correctly staged.

In particular, hm9k throws away information it receives about applications that are not staged [1]. That means that when the CC asks hm9k for information about the app [2], it has none and returns '-1' for instances (and all other metrics).

Basically, it looks like the summary endpoint or `instances_reporter.rb` [3] should be evaluating the package state a little more carefully so it can present the correct information as hm9k will not have info on apps that did not successfully stage.

[1]: https://github.com/cloudfoundry/hm9000/blob/17b3830948ded967bd4052d4975683e107a43030/desiredstatefetcher/desired_state_fetcher.go#L162
[2]: https://github.com/cloudfoundry/cloud_controller_ng/blob/4b23dd32b39174d5bf1c11170851c2ea99704272/app/controllers/runtime/app_summaries_controller.rb#L18
[3]: https://github.com/cloudfoundry/cloud_controller_ng/blob/4b23dd32b39174d5bf1c11170851c2ea99704272/lib/cloud_controller/dea/instances_reporter.rb#L14-L17

**Comment by dieucao** [comment]
I believe this is addressed via https://github.com/cloudfoundry/cloud_controller_ng/pull/408
and https://github.com/cloudfoundry/cloud_controller_ng/pull/409
